tuna/worker_interface.py

```diff
@@ -315,18 +315,19 @@ def update_fdb_entry(self, session, solver):
     obj, fdb_entry = self.get_fdb_entry(session, solver)
     if obj:  # existing entry in db
       # This can be removed if we implement the delete orphan cascade
-      for blob in obj.blobs:
-        session.delete(blob)
       fdb_entry = obj
+      session.query(
+          self.dbt.kernel_cache).filter(self.dbt.kernel_cache.kernel_group ==
+                                        fdb_entry.kernel_group).delete()
```

Comment:
whats the reasoning for this delete?

Reply:
The kernel group is about to be replaced. The code this replaces also deletes the blobs.